pbs_light.go

```diff
@@ -313,10 +321,13 @@ func (deps *auctionDeps) auction(w http.ResponseWriter, r *http.Request, _ httpr
 			ametrics.RequestMeter.Mark(1)
 			accountAdapterMetric.RequestMeter.Mark(1)
 			if pbs_req.App == nil {
-				uid, _, _ := pbs_req.Cookie.GetUID(ex.FamilyName())
+				// If exchanges[bidderCode] exists, then deps.syncers[bidderCode] should exist too.
+				// The unit tests guarantee it.
```

Comment:
Can you help me identify the unit test?

`exchanges` is a map of its own and `NewSyncerMap()` creates a map again. This works best when the two of them don't interact with each other. I only wonder if this is always true:

> If exchanges[bidderCode] exists, then deps.syncers[bidderCode] should exist too.

Reply:
After looking into it... you're totally right, and this is a bug on my part.

`districtm` is part of the `exchange`, but not the `NewSyncerMap()`. `TestSyncers` inside `usersync/usersync_test.go` makes sure that the `NewSyncerMap()` is maintained with `openrtb_ext.BidderMap`, but there's no analogous test for the `exchange`. This comment is also a lie, because `exchange` includes `districtm`, but the `BidderMap` and `Usersyncers` don't.

I'll find a way to hack the legacy code and add a new test for this.

Reply:
All fixed ^^. It's very hacky, but... at least the hacks are limited to the legacy code.